crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -384,60 +389,183 @@ pub fn extract_skinned_meshes(
     commands.insert_or_spawn_batch(values);
 }
 
+/// Data necessary to be equal for two draw commands to be mergeable
+///
+/// This is based on the following assumptions:
+/// - Only entities with prepared assets (pipelines, materials, meshes) are
+///   queued to phases
+/// - View bindings are constant across a phase for a given draw function as
+///   phases are per-view
+/// - `prepare_mesh_uniforms` is the only system that performs this batching
+///   and has sole responsibility for preparing the per-object data. As such
+///   the mesh binding and dynamic offsets are assumed to only be variable as a
+///   result of the `prepare_mesh_uniforms` system, e.g. due to having to split
+///   data across separate uniform bindings within the same buffer due to the
+///   maximum uniform buffer binding size.
+#[derive(Default, PartialEq, Eq)]
+struct BatchMeta<'mat, 'mesh> {
+    /// The pipeline id encompasses all pipeline configuration including vertex
+    /// buffers and layouts, shaders and their specializations, bind group
+    /// layouts, etc.
+    pipeline_id: Option<CachedRenderPipelineId>,
```

Comment:
why `Option` for most of these?  it looks unnecessary.

Reply:
Good point. I think there was a reason at some point but I guess it's been coded away. Fixing.

Reply:
And I rediscovered the reason. There needs to be some uninitialised state at the beginning of the phase to be able to start a new batch for the first batch. There is probably another way to do it, if some ids can be defined as invalid. DrawFunctionId could be NonMaxU32 and then this would be Option<NonMaxU32>. For some reason CachedPipelineId is a usize for some reason (probably indexing?) but they do have some `pub const INVALID: Self = CachedRenderPipelineId(usize::MAX);` but should probably also be a NonMaxU32. And the Handle<Mesh> should be a HandleId which has a way of being defined as an invalid value. After moving stuff to NonMaxU32, they would still be used as Option<NonMaxU32> here though.

Reply:
… I can’t see the wood for the trees.

Reply:
Done.